### DISCUSSION TITLE
Fooocus using only 6gb of 10 available of gpu ram

### DISCUSSION THREAD

**Discussion by doomkugar** [lllyasviel/Fooocus#2605] [Q&A]
As I wrote in the title, I'm using Fooocus with an Nvidia 3080 with 10GB of RAM. When I launch the process, from performance monitoring, I see that only 6 out of 10 are being used. Is there something wrong?
![low ram](https://github.com/lllyasviel/Fooocus/assets/20595814/0715e596-e359-41d3-8d3a-01973dde0ece)

**Comment by poor7** [comment]
Can I see the console log? How much VRAM is determined there?

**Reply by doomkugar** [reply]
D:\Fooocus>.\python_embeded\python.exe -s Fooocus\entry_with_update.py --listen --port 7860
Already up-to-date
Update succeeded.
[System ARGV] ['Fooocus\\entry_with_update.py', '--listen', '--port', '7860']
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
Fooocus version: 2.3.1
[Cleanup] Attempting to delete content of temp dir C:\Users\Admin\AppData\Local\Temp\fooocus
[Cleanup] Cleanup successful
Total VRAM 10240 MB, total RAM 32688 MB
Set vram state to: NORMAL_VRAM
Always offload VRAM
Device: cuda:0 NVIDIA GeForce RTX 3080 : native
VAE dtype: torch.bfloat16
Using pytorch cross attention
Refiner unloaded.
Running on local URL:  http://0.0.0.0:7860
model_type EPS
UNet ADM Dimension 2816

To create a public link, set `share=True` in `launch()`.
Using pytorch attention in VAE
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
Using pytorch attention in VAE
extra {'cond_stage_model.clip_g.transformer.text_model.embeddings.position_ids', 'cond_stage_model.clip_l.logit_scale', 'cond_stage_model.clip_l.text_projection'}
Base model loaded: D:\Automatic1111 Stable Diffusion\stable-diffusion-webui\models\Stable-diffusion\XL\realisticFreedomSFW_wonderland.safetensors
Request to load LoRAs [['add_detail.safetensors', 0.1], ['xl_more_art-full_v1.safetensors', 0.1], ['None', 1.0], ['None', 1.0], ['None', 1.0]] for model [D:\Automatic1111 Stable Diffusion\stable-diffusion-webui\models\Stable-diffusion\XL\realisticFreedomSFW_wonderland.safetensors].
Loaded LoRA [D:\Automatic1111 Stable Diffusion\stable-diffusion-webui\models\Lora\xl_more_art-full_v1.safetensors] for UNet [D:\Automatic1111 Stable Diffusion\stable-diffusion-webui\models\Stable-diffusion\XL\realisticFreedomSFW_wonderland.safetensors] with 788 keys at weight 0.1.
Fooocus V2 Expansion: Vocab with 642 words.
Fooocus Expansion engine loaded for cuda:0, use_fp16 = True.
Requested to load SDXLClipModel
Requested to load GPT2LMHeadModel
Loading 2 new models
[Fooocus Model Management] Moving model(s) has taken 0.61 seconds
Started worker with PID 12584
App started successful. Use the app with http://localhost:7860/ or 0.0.0.0:7860
[Parameters] Adaptive CFG = 7
[Parameters] Sharpness = 2
[Parameters] ControlNet Softness = 0.25
[Parameters] ADM Scale = 1.5 : 0.8 : 0.3
[Parameters] CFG = 7.0
[Parameters] Seed = 2843574891432191242
[Parameters] Sampler = dpmpp_2m_sde_gpu - karras
[Parameters] Steps = 30 - 15
[Fooocus] Initializing ...
[Fooocus] Loading models ...
Refiner unloaded.
[Fooocus] Processing prompts ...
[Fooocus] Encoding positive #1 ...
[Fooocus Model Management] Moving model(s) has taken 0.20 seconds
[Fooocus] Encoding negative #1 ...
[Parameters] Denoising Strength = 1.0
[Parameters] Initial Latent shape: Image Space (1024, 1024)
Preparation time: 0.69 seconds
[Sampler] refiner_swap_method = joint
[Sampler] sigma_min = 0.0291671771556139, sigma_max = 14.614643096923828
Requested to load SDXL
Loading 1 new model
[Fooocus Model Management] Moving model(s) has taken 4.68 seconds
100%|██████████████████████████████████████████████████████████████████████████████████| 30/30 [00:11<00:00,  2.69it/s]
Requested to load AutoencoderKL
Loading 1 new model
[Fooocus Model Management] Moving model(s) has taken 0.18 seconds
Calculating sha256 for D:\Automatic1111 Stable Diffusion\stable-diffusion-webui\models\Stable-diffusion\XL\realisticFreedomSFW_wonderland.safetensors: 3dfe2ed3a99b44a0efe96ba1e2cbd4f422bd36ea7c0a294af0e1b3124d790933
Calculating sha256 for D:\Automatic1111 Stable Diffusion\stable-diffusion-webui\models\Lora\add_detail.safetensors: 47aaaf0d2945ca937151d61304946dd229b3f072140b85484bc93e38f2a6e2f7
Calculating sha256 for D:\Automatic1111 Stable Diffusion\stable-diffusion-webui\models\Lora\xl_more_art-full_v1.safetensors: 15e31fe2b6ae2e77ee47a3ccdf27bd14f7b54ce27c6a58502875fdad26f34460
Image generated with private log at: D:\Fooocus\Fooocus\outputs\2024-03-23\log.html
Generating and saving time: 33.38 seconds
Requested to load SDXLClipModel
Requested to load GPT2LMHeadModel
Loading 2 new models
Total time: 34.09 seconds
[Fooocus Model Management] Moving model(s) has taken 0.79 seconds

**Reply by poor7** [reply]
`--always-high-vram`  the key should completely occupy all 10 GB.

**Reply by doomkugar** [reply]
and it will speed up generation time?

**Reply by poor7** [reply]
> and it will speed up generation time?

It might speed it up a little.

**Reply by mashb1t** [reply]
No, it wil not, but it will decrease loading time as the model is kept in VRAM between generations, as mentioned before. The generation speed is still limited by your system performance, mostly GPU.

**Comment by mashb1t** [comment]
@doomkugar there is nothing wrong this is even a good thing as Fooocus is optimized to use as little as possible VRAM for maximum compatibility with also low-tier hardware.
Check out the arguments in the readme if you like to further decrease loading time inbetween generations.

**Reply by doomkugar** [reply]
i7 8700, 32gb ram,nvidia 3080, all loaded on nvme.. ssd is so low tier?

**Reply by mashb1t** [reply]
This is not what i am implying, Fooocus simply doesn't need more VRAM to use the potential of Stable Diffusion to full extend. The bottleneck still is your GPU performance.
